plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -0,0 +1,197 @@
+package io.jenkins.plugins.casc;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class SchemaGeneration {
+
+    final static JSONObject schemaTemplateObject = new JSONObject()
+        .put("$schema", "http://json-schema.org/draft-07/schema#")
+        .put("description", "Jenkins Configuration as Code")
+        .put("type", "object");
+
+    public static JSONObject generateSchema() {
+
+        /**
+         * The initial template for the JSON Schema
+         */
+        JSONObject schemaObject = new JSONObject(schemaTemplateObject.toString());
+        /**
+         * This generates the schema for the root configurators
+         * Iterates over the root elements and adds them to the schema.
+         */
+        schemaObject.put("properties", generateRootConfiguratorObject());
+        /**
+         * This generates the schema for the base configurators
+         * Iterates over the base configurators and adds them to the schema.
+         */
+        JSONObject schemaConfiguratorObjects = new JSONObject();
+        ConfigurationAsCode configurationAsCodeObject = ConfigurationAsCode.get();
```

Comment:
I would shorten the name object is unnecessary

Reply:
Considered +1